components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/utils/mappings/SubscriptionMappingUtil.java

```diff
@@ -90,6 +87,60 @@ public static SubscriptionListDTO fromSubscriptionListToDTO(List<SubscribedAPI>
         return subscriptionListDTO;
     }
 
+    /**
+     * Converts a List object of SubscribedAPIs into a DTO
+     *
+     * @param subscriptions a list of SubscribedAPI objects
+     * @param query query to filter subscriptions
+     * @return SubscriptionListDTO object containing SubscriptionDTOs
+     */
+    public static SubscriptionListDTO fromSubscriptionListToDTO(List<SubscribedAPI> subscriptions, String query)
+            throws APIManagementException {
+        SubscriptionListDTO subscriptionListDTO = new SubscriptionListDTO();
+        List<SubscriptionDTO> subscriptionDTOs = subscriptionListDTO.getList();
+        if (subscriptionDTOs == null) {
+            subscriptionDTOs = new ArrayList<>();
```

Comment:
can't we move this to L100? and then assign it to  
subscriptionDTOs = subscriptionListDTO.getList();

Reply:
No, What this is it takes a reference to the List<SubscriptionDTO> of subscriptionListDTO obj, and as a precaution initialize it if it is null.